test/support/integration_test.rb

```diff
@@ -16,6 +16,10 @@ def initialize(args)
   end
 
   def setup
+    # search_config is a global object that has state, so make sure it's reset
```

Comment:
Yeah definitely. I'm hoping that by just making the dependencies on it more explicit, it will make further refactoring easier.

For now I think it's useful for *something* to have knowledge about all the data sources (content indexes, metasearch index, spelling index, registries) since before you had to dig into the query building code to understand where data was coming from.

I think we will be able to remove some of the complexity once we start rearchitecting the indexing. :crossed_fingers: